### PR TITLE
Provide "Did you mean? task_name" suggestions if task was not found

### PR DESCRIPTION
**Pull request by yuki24**
similar to #19, but also suggests task names based on the edit distance.

### CODE REVIEW

File: lib/rake/task_manager.rb [hunk 1]
```diff
@@ -301,6 +321,33 @@ def get_description(task)
       desc
     end
 
+    # Return a value representing the "cost" of transforming str1 into str2
+    def levenshtein_distance(str1, str2)
```
Comment (zzak) [inline lib/rake/task_manager.rb]: We pass this method around, but I think it should be public in RubyGems, no?

Comment (yuki24) [inline lib/rake/task_manager.rb]: It is. We can just `include Gem::Text`and call `#levenshtein_distance`. I think you already know it had a bug, but maybe we shouldn't worry about it?

Comment (zzak) [inline lib/rake/task_manager.rb]: On the other hand, Rake seems to keep a safe distance from RubyGems (maybe for a reason).

This _could_ introduce some circular dependencies by depending on Gem itself.. and I'm not seeing anywhere besides `testtask` that uses it.

One could argue this method should be in a separate gem that both can require, and other people would find it useful. If that is the case, why not move it to core?

Comment (yuki24) [inline lib/rake/task_manager.rb]: Yeah, I've been thinking about it - providing `levenshtein_distance` as a stdlib of Ruby. Rubygems and Bundler already (and possibly Rake) use the Levenshtein algorithm so I think it's a good idea to move it to core. I'll file a feature request to https://bugs.ruby-lang.org.

That being said, I realized that rake still needs to have support for 1.8.7, meaning we can't just use fresh features introduced recently. Should we just not merge this for now? I don't feel comfortable forcing others to maintain code that is copy-and-pasted...

Comment (zzak) [inline lib/rake/task_manager.rb]: We should avoid duplication whenever possible

Comment (yuki24) [inline lib/rake/task_manager.rb]: filed a ticket: https://bugs.ruby-lang.org/issues/10882

Comment (drbrain) [inline lib/rake/task_manager.rb]: I don't see why we can't merge this now while we wait for ruby to have levenshtein distance.

Comment (zzak) [inline lib/rake/task_manager.rb]: If we merge this now I think we need to consider using `Gem::Text` instead of copy&paste

Comment (drbrain) [inline lib/rake/task_manager.rb]: The necessary methods for `Gem::Text` don't exist for all versions of RubyGems that rake supports.

**Comment by yuki24** [comment]
closing this pull request as there's an ongoing discussion on https://bugs.ruby-lang.org/issues/11252. We might be able to use some of the features of `did_you_mean` in the near future, but I guess it's still too early to have this conversation here.

**Pull request closed without merging**